### PR TITLE
New objectives are returned in ChannelResults

### LINKED ISSUE #3228: Wallet emits "OperationStarted" & "OperationSucceeded"
# Why: Problem description :confused:

The ChannelManager currently has trouble managing channels -- it has trouble handling process crashes and network errors.

Much more motivation can be found [here](https://www.notion.so/RCF-1-First-class-objectives-2a26247ce3244b8aafa4f9397df8465f#187917c1a0d54dd3bf9715f8415f65f2).

# What: Describe the solution you'd like :sparkles:  
1. When the wallet begins a "top-level" objective for a channel, it should emit a "OperationStarted" event.
2. When the wallet successfully completes a "top-level" objective for a channel, it should emit a `"OperationSucceeded"` event.
3. In addition, wallet API calls that create new objectives should include a handle of the newly created objectives.

# Describe alternatives you've considered :thinking:
See [the section from the RFC](https://www.notion.so/RCF-1-First-class-objectives-2a26247ce3244b8aafa4f9397df8465f#8512ecba8e234575a852d0028b29fb58).

**Additional context**
[RFC 1](https://www.notion.so/e2a4c4b6c68d4d7693667687c54b416b?v=20a84ab82c284e5a92863fdb000f9b4a&p=2a26247ce3244b8aafa4f9397df8465f).

### PR DESCRIPTION
**Pull request by andrewgordstewart**
## Description
createdObjectives are stored in a WalletResponse, and then returned in Single/MultipleChannelOutput to the app.

Fixes #3228, with a slightly expanded scope: it's important for the app to learn about objectives created as a direct result of an API call.

## Alternatives
I chose not to modify the wallet API, besides adding optional `newObjectives` to `Single/MultipleChannelResult`, because it seemed outside the scope of this PR.

### Code quality
9/9

### CODE REVIEW

**Comment by andrewgordstewart** [comment]
Force pushed to deal with Squash & Merge conflicts.

File: packages/server-wallet/src/wallet/wallet-response.ts [hunk 1]
```diff
@@ -244,13 +252,14 @@ export class WalletResponse {
     outputs: (SingleChannelOutput | MultipleChannelOutput)[]
   ): MultipleChannelOutput {
     const channelResults = mergeChannelResults(
-      outputs
-        .map(m => (isSingleChannelMessage(m) ? [m.channelResult] : m.channelResults))
-        .reduce((cr1, cr2) => cr1.concat(cr2))
+      outputs.flatMap(m => (isSingleChannelMessage(m) ? [m.channelResult] : m.channelResults))
```
Comment (snario) [inline packages/server-wallet/src/wallet/wallet-response.ts:255]: 👍

File: packages/server-wallet/src/wallet/wallet-response.ts [hunk 2]
```diff
@@ -207,6 +213,7 @@ export class WalletResponse {
     return {
       outbox: mergeOutgoing(this.outbox),
       channelResult: this.channelResults[0],
+      newObjective: this.createdObjectives[0],
       // objectivesToApprove: this.objectivesToApprove, // TODO: re-enable
```
Comment (snario) [inline packages/server-wallet/src/wallet/wallet-response.ts:217]: This line is probably unnecessary / outdated at this point and can be removed reasonably within the scope of this PR.

**Review by snario — APPROVED** [review]

File: packages/server-wallet/src/wallet/wallet-response.ts (hunk 2 above)
Comment (andrewgordstewart) [reply packages/server-wallet/src/wallet/wallet-response.ts:217]: Hmm yeah, I had no idea what to make of this comment.

It seems to as if we need to figure out how "objective approval" needs to work. But I'm happy to remove these TODOs -- they don't seem to add any value anymore.

**Pull request merged**